com.unity.ml-agents.extensions/Runtime/Sensors/ArticulationHierarchyUtil.cs

```diff
@@ -0,0 +1,60 @@
+#if UNITY_2020_1_OR_NEWER
+
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Unity.MLAgents.Extensions.Sensors
+{
+
+    public class ArticulationHierarchyUtil : HierarchyUtil
```

Comment:
I don't think this is a Util class. ArticulationHierarchyHelper ?

Reply:
genuinely curious, what's the difference between a util and a helper class?

Reply:
i agree that this seems less like a utility class as it has state, and can be instantiated.  That goes for HierarchyUtil as well.

Reply:
Yeah, I don't get util vs helper either. They're both too generic.

Now that I switched to Pose (instead of writing my own struct), what about PoseCollection or PoseGenerator?

Reply:
This class generates Poses from an Articulated body. The Articulated body is passed at construction. 
PoseExtrator / PoseReader / PoseStatus ?

Reply:
+1 to PoseExtrator (RigidBodyPoseExtrator, ArticulationBodyPoseExtrator for the implementations)